sw/device/lib/base/log.h

```diff
@@ -89,11 +89,17 @@ void base_log_internal_dv(log_severity_t severity, const char *format, ...);
 // "" foo "" is a common C idiom to assert that a macro parameter is a string.
 #define LOG(severity, format, ...)                                       \
   do {                                                                   \
-    /* The |false| below will eventually be replaced with a device.h     \
-       function |device_is_dv()| or similar, which determines if the     \
-       current device is a DV testbench. */                              \
-    if (false) {                                                         \
-      base_log_internal_dv(severity, "" format "", ##__VA_ARGS__);       \
+    if (kDeviceType == kDeviceSimDV) {                                   \
+      /* Prepend format string with additional info - severity, file     \
+       * name, line and the number of var args.                          \
+       */                                                                \
+      static const char log[] __attribute__((section(".logs"))) =        \
+        "" severity "" " \"" "" __FILE__ "" "\" "                        \
+        LOG_DV_STRINGIFY(__LINE__) " "                                   \
+        LOG_DV_STRINGIFY(VA_NARGS(format, ##__VA_ARGS__))                \
+        " \"" "" format "" "\"";                                         \
```

Comment:
Why are all these things smushed together into one string? Wouldn't it make sense to have multiple regions on the DV side, for writing the severity, filename, line, and format, individually? The whole reason I created log_severity_t was to avoid this problem.